docs/rpc-endpoints.md

```diff
@@ -0,0 +1,255 @@
+                "buffer": {
+                  "length": 32
+                }
+              }
+            },
+            {
+              "name": "burn-block-time",
+              "type": "uint128"
+            },
+            {
+              "name": "id-hash",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            },
+            {
+              "name": "stacks-hash",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            },
+            {
+              "name": "stacks-miner",
+              "type": "principal"
+            },
+            {
+              "name": "vrf-seed",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            }
+          ]
+        }
+      }
+    }
+  ],
+  "variables": [],
+  "maps": [
+    {
+      "name": "block-data",
+      "key": [
+        {
+          "name": "height",
+          "type": "uint128"
+        }
+      ],
+      "value": [
+        {
+          "name": "btc-hash",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        },
+        {
+          "name": "burn-block-time",
+          "type": "uint128"
+        },
+        {
+          "name": "id-hash",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        },
+        {
+          "name": "stacks-hash",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        },
+        {
+          "name": "stacks-miner",
+          "type": "principal"
+        },
+        {
+          "name": "vrf-seed",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        }
+      ]
+    }
+  ],
+  "fungible_tokens": [],
+  "non_fungible_tokens": []
+}
+```
+
+### GET /v2/contracts/source/[Stacks Address]/[Contract Name]
+
+Fetch the source for a smart contract. Returned as a JSON string.
+
+### POST /v2/contracts/call-read/[Stacks Address]/[Contract Name]/[Function Name]
+
+Call a read-only public function on a given smart contract.
+
+The smart contract and function are specified using the URL path. The arguments and
+the simulated `tx-sender` are supplied via the POST body in the following JSON format:
+
+```
+{
+  "sender": "SP31DA6FTSJX2WGTZ69SFY11BH51NZMB0ZW97B5P0.get-info",
+  "arguments": [ "0011...", "00231..." ]
+}
+```
+
+Where sender is either a Contract identifier or a normal Stacks address, and arguments
+is an array of hex serialized Clarity values.
+
+This endpoint returns a JSON object of the following form:
+
+```
+{
+  "okay": true,
+  "result": "0011..."
+}
+```
+
+Where `"okay"` is `true` if the function executed successfully, and result contains the
+hex serialization of the Clarity return value.
```

Comment:
What's our take on conventionally prefixing hexa strings with `0x` in these API?

Reply:
I don't think it's necessary.

Reply:
That would work for me.

Reply:
The `0x` prefix helpful due to how integers tend to be represented in JSON. 
E.g. when you see `"3039"` is this hex for `12345` just a string integer?

Reply:
Yeah -- I agree, the hex prefix makes it way more clear what the string is representing.

Reply:
When you put it that way, I also agree that the `0x` prefix will be good :+1:

Reply:
Updated in 8fa73b47f -- this also changed the output for the balance to a hex string (since a u128 will stress JSON integer parsers in unexpected, but usually terrible, ways).